src/Raven.Server/Documents/Replication/ResolveConflictOnReplicationConfigurationChange.cs

```diff
@@ -400,6 +412,125 @@ public DocumentConflict ResolveToLatest(List<DocumentConflict> conflicts)
 
             return latestDoc;
         }
+
+        private void ResolveAttachmentsConflicts(DocumentsOperationContext context, LazyStringValue lowerIdString, DocumentConflict resolved, DocumentConflict conflict, Document local)
+        {
+            using var scope = Slice.External(context.Allocator, lowerIdString, out var lowerId);
+            if (resolved.Doc.TryGet(Constants.Documents.Metadata.Key, out BlittableJsonReaderObject metadata) == false
+                || metadata.TryGet(Constants.Documents.Metadata.Attachments, out BlittableJsonReaderArray attachments) == false)
+                return;
+
+            var currentDocumentAttachments = (from BlittableJsonReaderObject bjro in attachments select JsonDeserializationClient.AttachmentName(bjro)).ToList();
+            var storageAttachmentsDetails = _database.DocumentsStorage.AttachmentsStorage.GetAttachmentDetailsForDocument(context, lowerId);
+
+            var oldAttachmentsMetadata = ExtractOldAttachmentsMetdataIfNeeded(attachments.Length, new List<BlittableJsonReaderObject> { local.Data, conflict.Doc });
+            var attachmentsToRemove = oldAttachmentsMetadata?.Select(x => x.Name).ToList();
+
+            foreach (var currentAttachment in currentDocumentAttachments)
+            {
+                var count = CountInAttachmentsList(currentAttachment.Name, storageAttachmentsDetails);
+                switch (count)
+                {
+                    case 1:
+                        RemoveFromAttachmentsToRemoveListIfNeeded(currentAttachment.Name);
+                        continue;
+                    case 0:
+                        // attachment exists in metadata, does not exist in storage (renamed/added in conflict resolution)
+                        // try to find attachments in attachment storage with both same hash and content type
+                        var matches = storageAttachmentsDetails.Where(x => x.Hash == currentAttachment.Hash && x.ContentType == currentAttachment.ContentType).ToList();
+                        if (matches.Count == 0)
+                        {
+                            // attachment is in the metadata but doesn't exist in the attachments storage
+                            throw new AttachmentConflictException($"Attachment '{currentAttachment.Name}' was found in '{nameof(DocumentFlags.Resolved)}' document '{lowerId}' metadata, but was not found in the attachments storage using the following: Hash: '{currentAttachment.Hash}', ContentType: '{currentAttachment.ContentType}'.", lowerId.ToString());
```

Comment:
Shouldn't this be a missing attachment error?

Reply:
This is to prevent from adding an attachment in conflict resolver script
https://github.com/ravendb/ravendb/pull/11332/files#diff-0274c68798c060bc7f04679d2b76c93d9d1f92404872601fcf7d4e7f547b224fR1951